iocore/eventsystem/UnixEThread.cc

```diff
@@ -142,6 +142,97 @@ EThread::process_event(Event *e, int calling_code)
   }
 }
 
+void EThread::process_queue(Que(Event, link) * NegativeQueue)
+{
+  Event *e;
+
+  // Move events from the external thread safe queues to the local queue.
+  EventQueueExternal.dequeue_external();
+
+  // execute all the available external events that have
+  // already been dequeued
+  while ((e = EventQueueExternal.dequeue_local())) {
+    if (e->cancelled) {
+      free_event(e);
+    } else if (!e->timeout_at) { // IMMEDIATE
+      ink_assert(e->period == 0);
+      process_event(e, e->callback_event);
+    } else if (e->timeout_at > 0) { // INTERVAL
+      EventQueue.enqueue(e, cur_time);
+    } else { // NEGATIVE
+      Event *p = nullptr;
+      Event *a = NegativeQueue->head;
+      while (a && a->timeout_at > e->timeout_at) {
+        p = a;
+        a = a->link.next;
+      }
+      if (!a) {
+        NegativeQueue->enqueue(e);
+      } else {
+        NegativeQueue->insert(e, p);
+      }
+    }
+  }
+}
+
+void
+EThread::execute_regular()
+{
+  Event *e;
+  Que(Event, link) NegativeQueue;
+  ink_hrtime next_time = 0;
+
+  // give priority to immediate events
+  for (;;) {
+    if (unlikely(shutdown_event_system == true)) {
+      return;
+    }
+
+    process_queue(&NegativeQueue);
+
+    bool done_one;
+    do {
+      done_one = false;
+      // execute all the eligible internal events
+      EventQueue.check_ready(cur_time, this);
+      while ((e = EventQueue.dequeue_ready(cur_time))) {
+        ink_assert(e);
+        ink_assert(e->timeout_at > 0);
+        if (e->cancelled)
+          free_event(e);
+        else {
+          done_one = true;
+          process_event(e, e->callback_event);
+        }
+      }
+    } while (done_one);
+
+    // execute any negative (poll) events
+    if (NegativeQueue.head) {
+      process_queue(&NegativeQueue);
+
+      // execute poll events
+      while ((e = NegativeQueue.dequeue())) {
+        process_event(e, EVENT_POLL);
+      }
+    }
+
+    next_time             = EventQueue.earliest_timeout();
+    ink_hrtime sleep_time = next_time - Thread::get_hrtime_updated();
+    if (sleep_time > 0) {
+      sleep_time = std::min(sleep_time, HRTIME_MSECONDS(THREAD_MAX_HEARTBEAT_MSECONDS));
+    } else {
+      sleep_time = 0;
+    }
+
+    if (n_ethreads_to_be_signalled) {
```

Comment:
The `n_ethreads_to_be_signalled` is always 0.

Reply:
Yes. That is by design. The `ethreads_to_be_signalled` is obsolete now and should probably be removed. In effect this was changed always "fast signal" along the lines of the code at lines 61..65. If that path is taken in the current code then `n_ethreads_to_be_signalled` will always be in the same way. The real change here is to force `fast_signal` to always be `true`, making this behavior one that is entirely possible in the old code.

Reply:
Why don't you remove these 3 lines since the if-condition is always false and the `flush_signals()` never be called ?